### DISCUSSION TITLE
The type or namespace name 'BepInEx' could not be found (are you missing a using directive or an assembly reference?)

### DISCUSSION THREAD

**Discussion by LindaLawton** [BepInEx/BepInEx#298] [Help and Q&A]
I have been trying to build my first mod for two days now.   Previously I was following others tutorials and I was unable to get them to compile. 

Now I have tired to follow yours line for line.  [Writing a basic plug-in](https://docs.bepinex.dev/articles/dev_guide/plugin_tutorial/2_plugin_start.html)  this is what i get when i try to build it.

>  ExamplePlugin.cs(1, 7): [CS0246] The type or namespace name 'BepInEx' could not be found (are you missing a using directive or an assembly reference?)
  ExamplePlugin.cs(7, 34): [CS0246] The type or namespace name 'BaseUnityPlugin' could not be found (are you missing a using directive or an assembly reference?)
  ExamplePlugin.cs(6, 6): [CS0246] The type or namespace name 'BepInPluginAttribute' could not be found (are you missing a using directive or an assembly reference?)
  ExamplePlugin.cs(6, 6): [CS0246] The type or namespace name 'BepInPlugin' could not be found (are you missing a using directive or an assembly reference?)

I can see that the dlls are there. in the csproj file

 ><ItemGroup>
        <Reference Include="Assembly-CSharp, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null">
          <HintPath>..\Libs\Assembly-CSharp.dll</HintPath>
        </Reference>
        <Reference Include="BepInEx, Version=5.4.15.0, Culture=neutral, PublicKeyToken=null">
          <HintPath>..\Libs\BepInEx.dll</HintPath>
        </Reference>
        <Reference Include="System" />
        <Reference Include="System.Core" />
        <Reference Include="System.Xml" />
        <Reference Include="UnityEngine">
            <HintPath>C:\Program Files\Unity\Hub\Editor\2020.3.16f1\Editor\Data\Managed\UnityEngine.dll</HintPath>
        </Reference>
        <Reference Include="UnityEngine, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null">
          <HintPath>..\Libs\UnityEngine.dll</HintPath>
        </Reference>
        <Reference Include="UnityEngine.CoreModule, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null">
          <HintPath>..\Libs\UnityEngine.CoreModule.dll</HintPath>
        </Reference>
    </ItemGroup>

They also appear to be properly loaded

![image](https://user-images.githubusercontent.com/6592543/130739785-4a4385ba-573c-43c6-82ee-6f9540a12a5c.png)

It even shows in the code that it can find the dlls

![image](https://user-images.githubusercontent.com/6592543/130739849-244d328e-d4f5-4cde-a325-d4745ec54681.png)

It just simply cant find them when it tries to build.

Any help would be greatly apricated im doing something wrong here and it doesn't make any sense it should just be a matter of importing the reference dlls.

# what i have tried

Yes i tried to clean the solution.
Yes the references are set to Copy local

# .net 3.1 is installed

>.NET SDKs installed:
  3.1.412 [C:\Program Files\dotnet\sdk]
  5.0.300 [C:\Program Files\dotnet\sdk]

**Comment by IDontHaveIdea** [comment]
Hi,

What is the game you are trying to make the plugin for.

**Reply by LindaLawton** [reply]
[dyson sphere program](https://en.wikipedia.org/wiki/Dyson_Sphere_Program)


Since I cant edit the question because there is a message box i cant close over the submit button.

![image](https://user-images.githubusercontent.com/6592543/130748476-5dd07320-d2d6-4de7-80cb-4e7f17483e79.png)


# update


 >Microsoft.Common.CurrentVersion.targets(2123, 5): [MSB3258] The primary reference "BepInEx, Version=5.4.15.0, Culture=neutral, PublicKeyToken=null" could not be resolved because it has an indirect dependency on the .NET Framework assembly "mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" which has a higher version "4.0.0.0" than the version "2.0.0.0" in the current target framework.

Im using .net core 3.5.  This seams strange.  This doesn't need something silly like .Net 4.5 or something does it?

**Comment by LindaLawton** [comment]
I think I found the problem.  When i created the project I selected Unity Class Library as the project type.  Not just a plain class library. 

I wrongly assumed they would be the same thing.  They are not.   The unity class library project type is completely different then a standard class library.

It builds now.

**Comment by IDontHaveIdea** [comment]
Yes has it has to be a **.Net Framework Class Library**
